shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/dml/assignment/AssignmentSegment.java

```diff
@@ -36,5 +39,7 @@
     
     private final ColumnSegment column;
     
+    private List<ColumnSegment> columns = new LinkedList<>();
```

Comment:
What about combine column and columns.


```
@RequiredArgsConstructor
@Getter
public final class AssignmentSegment implements SQLSegment {
    
    private final int startIndex;
    
    private final int stopIndex;
    
    private final List<ColumnSegment> columns;
    
    private final List<ExpressionSegment> values;
}
```

Reply:
> What about combining the column and columns?

We can combine columns as a list but since the `AssignmentSegment` is used across the ShardingSphere project, we need to make changes in all `AssignmentSegment` occurrences. For example in `EncryptAssignmentParameterRewriter`:

![AssignmentSegmentUseCases](https://user-images.githubusercontent.com/48581379/128676131-da81db17-f4c7-4e85-8b79-5a1b757a0af7.png)

What shall we do?

Reply:
> And change the value to list

I think we don't have to create a list for value, as an assignment can have only one value:

![UpdateSetClauseHasOnlyOneAssignmentValue](https://user-images.githubusercontent.com/48581379/128676925-cda31c6d-76b6-445f-beeb-1b23d3a8eb6c.png)

Please let me know what do you think.

Reply:
Below is not reasonable:
```
@RequiredArgsConstructor
@Getter
public final class AssignmentSegment implements SQLSegment {
    
    private final int startIndex;
    
    private final int stopIndex;
    
    private final ColumnSegment column;

    private List<ColumnSegment> columns = new LinkedList<>();
    
    private final ExpressionSegment value;
}
```
But what shall we do in this situation? @jingshanglu @tristaZero 
```
UPDATE employees a SET department_id = (SELECT department_id FROM departments WHERE location_id = '2100'), (salary, commission_pct) = (SELECT 1.1*AVG(salary), 1.5*AVG(commission_pct) FROM employees b WHERE a.department_id = b.department_id);
```

Reply:
@wgy8283335 @ThanoshanMV Maybe we can extract abstract class and combine these two idea.
```
public abstract class AssignmentSegment implements SQLSegment {
    
   List<ColumnSegment> getColumns();
   ExpressionSegment getValue();
}

@RequiredArgsConstructor
@Getter
public final class OracleAssignmentSegment extends AssignmentSegment {
    
   private final int startIndex;
    
    private final int stopIndex;

    private List<ColumnSegment> columns = new LinkedList<>();
    
    private final ExpressionSegment value;
}
```
```
SET department_id = (SELECT department_id FROM departments WHERE location_id = '2100')
expressed as
OracleAssignmentSegment{columns=[department_id],value=subquery}

(salary, commission_pct) = (SELECT 1.1*AVG(salary), 1.5*AVG(commission_pct) FROM employees b WHERE a.department_id = b.department_id)
expressed as 
OracleAssignmentSegment{columns=[salary, commission_pct],value=subquery}
```
Maybe it has better decoupling and scalability, but we need to make changes in all `AssignmentSegment` occurrences.

Reply:
@jingshanglu do you mean like this:

`AssignmentSegment`:

![AssignmentSegment](https://user-images.githubusercontent.com/48581379/129441421-217ab2bf-5f97-4899-9edf-adf425047d9e.png)

`OracleAssignmentSegment`:

![OracleAssignmentSegment](https://user-images.githubusercontent.com/48581379/129441447-bf2e0401-289d-4e4e-bc9d-b493941b3f09.png)

Reply:
Shall we create similar child classes for other SQL dialects, for example, `PostgreSQLAssignmentSegment` or shall we rename `OracleAssignmentSegment` to a generic one?

Currently, whenever we need to create an instance of `AssignmentSegment`, we have to utilize `OracleAssignmentSegment`. In the below image, the method `visitSetClause` is in `PostgreSQLStatementSQLVisitor` which returns an instance of `AssignmentSegment`:

![PostgreSQLIssue](https://user-images.githubusercontent.com/48581379/129445665-4b112d6b-56c8-4c85-afbf-98ad1ca02456.png)

Reply:
> Shall we create similar child classes for other SQL dialects, for example, `PostgreSQLAssignmentSegment` or shall we rename `OracleAssignmentSegment` to a generic one?
> 
> Currently, whenever we need to create an instance of `AssignmentSegment`, we have to utilize `OracleAssignmentSegment`. In the below image, the method `visitSetClause` is in `PostgreSQLStatementSQLVisitor` which returns an instance of `AssignmentSegment`:
> 
> ![PostgreSQLIssue](https://user-images.githubusercontent.com/48581379/129445665-4b112d6b-56c8-4c85-afbf-98ad1ca02456.png)

@ThanoshanMV Yes, maybe it is better to rename `OracleAssignmentSegment` to a genneric one,  define new subclasses when we need to expand.